src/python/genny/genny_auto_tasks.py

```diff
@@ -222,30 +271,18 @@ def construct_all_tasks_json():
             continue
 
         task_name = to_snake_case(base_parts[0])
-        t = c.task(task_name)
-        t.priority(5)  # The default priority in system_perf.yml
-
-        prepare_environment_vars = {
-            'test': task_name,
-            'auto_workload_path': fname
-        }
-
-        full_filename = '{}/src/workloads/{}'.format(get_project_root(), fname)
-        with open(full_filename, 'r') as handle:
-            try:
-                workload_dict = yaml.safe_load(handle)
-                autorun_spec = AutoRunSpec.create_from_workload_yaml(workload_dict)
-                if autorun_spec is not None and autorun_spec.prepare_environment_with is not None:
-                    prepare_environment_vars.update(autorun_spec.prepare_environment_with)
-            except Exception as e:
-                pass
-
-        t.commands([
-            CommandDefinition().function('prepare environment').vars(prepare_environment_vars),
-            CommandDefinition().function('deploy cluster'),
-            CommandDefinition().function('run test'),
-            CommandDefinition().function('analyze'),
-        ])
+        
+        prepare_environment_vars = get_prepare_environment_vars(task_name, fname)
+        
+        for prep_var in prepare_environment_vars:
+            t = c.task(prep_var['test'])
+            t.priority(5)  # The default priority in system_perf.yml
+            t.commands([
```

Comment:
see my recent PR just merged - I added a timeout here

Reply:
Tried to rebase; found out rebase doesn't play nice with PRs, merged, think it's good now.